### ISSUE TITLE
Flatpak hangs shutdown/reboot

### ISSUE THREAD

**Issue by cstoudt** [flatpak/flatpak#2637]
## Linux distribution and version
Operating system: Linux Mint 18.3 64-bit with linux kernel 4.4.0-141-generic
Hardware: MSI GP62 6700 laptop with Intel i7-6700HQ CPU

## Flatpak version
1.03

## Description of the problem
Flatpak hangs shutdown or reboot. In order to shutdown the laptop, a hard shutdown initiated with the power button is required.

The culprit appears to be the process flatpak-dbus-proxy.  If I manually kill that process before initiating reboot or shutdown then reboot/shutdown proceeds normally.

**Comment by matthiasclasen** [comment]
Are you running a flatpak app when that happens ?

**Comment by alexlarsson** [comment]
Hmm. Each flatpak-dbus-proxy is tied to a flatpak sandbox instance, and should be slaved to its lifetime. As soon as all the processes in that sandbox dies, then its pid1 dies and this should kill the proxy. It might be interesting to see what app the proxy came from.

However, its unclear why the existance of the proxy keeps us from shutting down. Its likely to be some process in the sandbox that did a dbus call (via the proxy) to keep the system alive....

**Comment by matthiasclasen** [comment]
Would be useful to see what `systemd-inhibit --list` says in in this situation.

**Comment by cstoudt** [comment]
The flatpak app SparkleShare is running in the background when I initiate a shutdown.  If I manually shut down SparkleShare and then initiate a shutdown, then shutdown proceeds normally.

Here's the output of `systemd-inhibit --list`

     Who: nom (UID 1000/nom, PID 1992/csd-power)
    What: sleep
     Why: Cinnamon needs to lock the screen
    Mode: delay

     Who: nom (UID 1000/nom, PID 1992/csd-power)
    What: handle-lid-switch
     Why: Multiple displays attached
    Mode: block

     Who: nom (UID 1000/nom, PID 2035/csd-media-keys)
    What: handle-power-key:handle-suspend-key:handle-hibernate-key
     Why: Cinnamon handling keypresses
    Mode: block

     Who: NetworkManager (UID 0/root, PID 1138/NetworkManager)
    What: sleep
     Why: NetworkManager needs to turn off networks
    Mode: delay

     Who: ModemManager (UID 0/root, PID 1065/ModemManager)
    What: sleep
     Why: ModemManager needs to reset devices
    Mode: delay

    5 inhibitors listed.

**Comment by alexlarsson** [comment]
So, this seems like its really a problem with sparkleshare, and the flatpak dbus proxy is just the middle man. There seem to be no related shutdown inhibitor registered though, which is weird.

@hbons Do you have any idea what is causing this?

**Comment by matthiasclasen** [comment]
@cstoudt are you seeing the same problem with any other flatpaks ?

**Comment by cstoudt** [comment]
No.  SparkleShare is the only flatpak installed.  Perhaps SparkleShare is to blame but I'm surprised that the flatpak sandbox environment doesn't catch this.

**Comment by hbons** [comment]
Very possible. :) @cstoudt can you open an issue on hbons/SparkleShare with version information please?

**Comment by truemaker** [comment]
This seems to be an issue with any flatpak that runs in the background.
